### DISCUSSION TITLE
What is the best practice regarding `__pypackages__/.gitignore`?

### DISCUSSION THREAD

**Discussion by lazarillo** [pdm-project/pdm#731] [Q&A] [answered]
I am *loving* PDM and getting rid of virtual environments forever.  :)

When first adding a package and therefore first creating a `__pypackages__` folder, the top level of the `__pypackages__` folder cleverly creates a `.gitignore` file that contains:

```
*
!.gitignore
```

Cool!  So everything in `__pypackages__` is already being ignored from the repo... *except the `.gitignore` file itself*.

But also PDM documentation says:

> PDM will write and store project-wide configurations in .pdm.toml and you are recommended to add following lines in the .gitignore:
> 
> ```
> .pdm.toml
> __pypackages__/
> ```

These recommendations are somewhat conflicting:  Once I add `__pypackages__` to the root `.gitignore`, then the subfolder `.gitignore` is ignored.

This isn't really a problem.  But I just want to standardize my team to a strategy.  Should we add `__pypackages__` to the root level `.gitignore`, or assume that the subfolder `.gitignore` will be the new standard going forward?  This makes only a slight difference in terms of the repo:  an empty folder called `__pypackages__` will be seen in, say, the repo on GitHub.

Thoughts?

**Comment by frostming** [comment, marked as answer]
[PEP 582](https://www.python.org/dev/peps/pep-0582/) says:

>Projects that use a source management system can include a `__pypackages__` directory (empty or with e.g. a file like `.gitignore`). After doing a fresh check out the source code, a tool like pip can be used to install the required dependencies directly into this directory.

It uses a **can**, so I suppose an empty `__pypackages__` can help tools understand it is a PEP 582 style project if there are any. It is changed sometime later, so the README might need an update.
